src/Illuminate/Routing/PendingResourceRegistration.php

```diff
@@ -149,6 +149,14 @@ public function parameter($previous, $new)
      */
     public function middleware($middleware)
     {
+        if (! is_string($middleware)) {
+            $middleware = Arr::wrap($middleware);
+
+            foreach ($middleware as $key => $value) {
```

Comment:
why not like this:
```php
foreach ($middleware as &$value) {
    $value = (string) $value;
 }
```

Reply:
Long time ago I read that PHP keeps the reference of the last value, and that foreach actually copies the array into the loop, and since PHP is optimized for copy-on-write, using the `$key` is more performant. Be free to correct me.